ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/Common.scala

```diff
@@ -0,0 +1,97 @@
+package com.daml.ledger.participant.state.kvutils.committing
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlLogEntry,
+  DamlStateKey,
+  DamlStateValue
+}
+import com.daml.ledger.participant.state.kvutils.KeyValueCommitting.Err
+
+object Common {
+  type DamlStateMap = Map[DamlStateKey, DamlStateValue]
+
+  /** A monadic computation that represents the process of committing which accumulates
+    * ledger state and finishes with the final state and a log entry.
+    * This is essentially State + Except.
+    */
+  final case class Commit[A](run: DamlStateMap => Either[CommitDone, (A, DamlStateMap)]) {
+    def flatMap[A1 >: A](f: A => Commit[A1]): Commit[A1] =
+      Commit { state =>
+        run(state) match {
+          case Left(done) => Left(done)
+          case Right((x, state2)) =>
+            f(x).run(state2)
+        }
+      }
+  }
+
+  /** The terminal state for the commit computation. */
+  final case class CommitDone(logEntry: DamlLogEntry, state: DamlStateMap)
+
+  object Commit {
+
+    /** Sequence commit actions which produces no intermediate values. */
+    def sequence[A](act: Commit[Unit], acts: Commit[Unit]*): Commit[Unit] = {
+      def go(
+          state: DamlStateMap,
+          act: Commit[Unit],
+          rest: Seq[Commit[Unit]]
+      ): Either[CommitDone, (Unit, DamlStateMap)] = {
+        act.run(state) match {
+          case Left(done) =>
+            Left(done)
+          case Right(((), state2)) =>
+            rest match {
+              case a +: as =>
+                go(state2, a, as)
+              case _ =>
+                Right(() -> state2)
+            }
+        }
+      }
+      Commit { state0 =>
+        go(state0, act, acts)
+      }
+    }
+
+    def run(act: Commit[Unit]): (DamlLogEntry, DamlStateMap) =
```

Comment:
Can we get comments on each of these building blocks? Right now only a handful are commented

Reply:
done.